### DISCUSSION TITLE
Using several private servers and port numbers at the same time

### DISCUSSION THREAD

**Discussion by julamp** [jamulussoftware/jamulus#1326] [Technical Support & Help]
I have tried to change the default port number 22124 to another by running jamulus server on terminal on Mac. I´m using private server on my home network. With any other number I can´t connect to my server . I also changed the port forwarding on router equal to different port number but didn´t help
Another question is also is it possible to run several servers (private) on same internal network? As if trying to have several Jamulus sessions running at the same time in the music school network?

**Comment by gilgongo** [comment]
Was there a reason why you changed from the default port number? If you do this then the clients that connect to it will also need to connect over that port.

On your second question, I think it's possible if you run each server on a separate port, and of course arrange for the clients to connect to them on those ports.

**Reply by julamp** [reply]
The reason to change port was to be able to run several private servers at the same network. Finally when tried different options on router firewall I did make it work when Service port number varies but local port number was always 22124 for all servers. Then clients could connect to different servers with service port number.  I just can not find the problem why changing the port number on Jamulus cuts the connectivity no matter what port number i then try give to client connection. I can´t either connect my own computer with 127.0.0.1 direct to server then. 
Can that be a problem on router?

**Reply by gilgongo** [reply]
In theory, if you set each jamulus server (using the `--port` option) to a different port, and have clients connecting to those servers using the adderess:port syntax in their connection window, and if you are port forwarding all those ports to the machine(s) the servers are running on then all should work.

I'm not entirely sure what the problem is that you are describing though. When you say "local port number was always 22124" can you explain that a bit more? Do you mean you see all clients connecting to the servers on that port? If so, perhaps there's some NAT issue going on?

> The reason to change port was to be able to run several private servers at the same network.

You may want to look at setting up a directory server for that and having clients connect to servers running on it.

**Reply by julamp** [reply]
Routers port forwarding has two ports for the IP , service port and local port.  What was strange that I put like for the first IP 22134 on service port and 22124 on local port, second IP 22234 on service port and 22124 on local port. Then  it worked with two different servers on different computers and clients could contact with public IP and service port number on right server.

**Comment by dcorson-ticino-com** [comment]
Were the servers all running on the same computer or on several computers on your network?
Remember to use 127.0.0.1:port for servers running on the same computer as the client.
I have done this many times and never had a problem.
I always opened a port forwarding window of about 32 ports to be able to play without resetting the forwarding on the router continuously.
Be sure to consult your router to have the right IP addresses on your network.
I can only imagine that the reason you see only port 22124 working is that only the computer using that port is really getting the right IP address.
Let us know when you have more clues.

**Reply by julamp** [reply]
In fact I got it work now. Forgot to use port number after 127.0.0.1   Then after the clients contact worked also with the same port number which I started jamulus server with and what was set on routers port forwarding.
Thank´s for advise
